### ISSUE TITLE
I'm buying boards and components, who will join?

### ISSUE THREAD

**Issue by eflukx** [schlae/snark-barker#5]
### TL;DR 

**We're (group) buying all parts for the SnarkBarker, that I will ship out @cost. There are 3 sets (of 10) left, so please let me know if you want to join in!**

--- 
_Original message_

Excuse me for spamming the issues section, but this might be of interest of people following this repo:

I have ordered (and received) 10 very nice bare SnarkBarker PCB's, but I only need one :) I'm planning to order rest of components soon to build my SnarkBarker, but happily order more parts for whomever interested.

If you're interested in joining up: I will ship PCB's + components @cost (expect around EUR50) across Europe (from Netherlands).

Let me know!

**Comment by northern** [comment]
I'll join. What's the total cost of a single board, i.e. PCB & components? I might be interested in a package for 2 or 3 depending on ~price~ total cost. I'm based in Australia.

Edit: Ah, OK. I was under the impression that 50 euro was shipping, just woke up :) I'll do 3 packages. Let me know if you can ship to AU. Exciting..!

**Comment by romsom** [comment]
Thanks for making the offer!
If I'm not too late, I'm in for 2 as well! :)

**Comment by raynet** [comment]
I would gladly take 1 if still available

**Comment by eflukx** [comment]
Hi Guys.. I see there's some interest.. Cool. :) Yep the offer is still up, haven't really got time do do something with this; really busy with some other important stuff.. 
But will get this moving soon. 

@northern  are you still in? Shipping to Australia will be ~50 AUD (30-35 eur, ex. taxes, import duties etc.)
@romsom  and @raynet where will you need the shipment?

I have the 10 boards here, can order parts this week when agreed on price...

**Comment by romsom** [comment]
@eflukx That's great news :)
In my case shipping would be to Germany.
Do you plan on buying pre-programmed 8051s?
I'm asking because I have access to a programmer and could easily program the 10 AT89S51s if that could reduce the price and/or effort to source the parts.

**Comment by northern** [comment]
@eflukx Hi there! Yes, still interested. Do you have an indication of total cost (without shipping), just to get an idea. @romsom brought up a good point regarding the pre-programmed 8051s. Could you elaborate on the what would be required on my end (as the receiver) other than soldering?

Thanks again for your efforts so far, much appreciated.

**Comment by raynet** [comment]
@eflukx  In my case shipping is to Finland.

**Comment by eflukx** [comment]
Ha.. I didn't think of the programming of the 8051.. :) I see Atmel has been so kind to include an ISP programming facility into this baby (reading the datasheet really is a trip down memory lane.. a uC with absolutely zilch peripherals.. soo cute.. 😸 ) I have some (actually quite a lot.. 😉 ) programmers of sorts around here, so I think I can manage to get you all a pre-programmed AT89S51! I dont think my "official" AtmelICE  will do the job, but it seems a $2 usbasp (with modified firmware) _will_ actually do the job.

I see we now have interest (including me) for about 7 kits. I'll get pricing for all of you by the end of this week. The volume knob I'll print for free :)

**Comment by romsom** [comment]
That's awesome, thank you very much! :)
Hehe, cheap tools with open/hackable firmware are the best! :)

**Comment by patricklaf** [comment]
Hello,
I am interested in one for France if possible.

**Comment by eflukx** [comment]
@patricklaf Ok.. put you on "the list" :)

**Comment by eflukx** [comment]
Hi guys, here's ~a price indication~ the final price:

Apart from shipping costs (and duties, where applicable), the sets will cost apx. 50 EUR when I order 10 sets:

- PCB cost ~ Eur 10
- Ebay (NOS) parts ~ Eur 6
- Mouser parts ~ Eur ~35~ 45
- 3D printer volume knob FREE :)
- IC sockets **not** included
- Blind ISA bracket included, please drill/dremel yourself.. :)

Shipping costs:

- Finland, Germany, France Eur 18,--
- Australia Eur 32,--

Payment preferably via PayPal (rogier_at_lodewijks_dot_org)
Payment up front please 💸 

Ebay -new old stock- parts are already ordered. PCB already here. When I have a minimum of 6 orders paid, I will order 10 sets at Mouser (volume potentiometer is back-ordered, so might take a little longer, or can ship without the knob). 

_edit: add shipping costs & "conditions"_
_edit: update Mouser price to include taxes_

**Comment by raynet** [comment]
Sounds great

**Comment by eflukx** [comment]
Pro tip: I _love_ the [KiCad iBOM plugin](https://github.com/openscopeproject/InteractiveHtmlBom), the interactive manual assembly helper tool
.
Attached the iBOM HTML export for the SnarkBarker: [snark_barker_ibom.zip](https://github.com/schlae/snark-barker/files/3374971/snark_barker_ibom.zip)

**Comment by romsom** [comment]
@eflukx You should just have received the payment for 2 boards (rounded up) :)

**Comment by eflukx** [comment]
> @eflukx You should just have received the payment for 2 boards (rounded up) :)

Yep, just got email from paypal, thanks!

**Comment by northern** [comment]
@eflukx I will transfer payment tonight when I come home (tomorrow morning for you). Can I email you on the email address that's on your Github profile to get your PayPal details? I'm in the run for 2 kits.

Edit: Oops. Just noticed you added your PayPal Id to a previous comment. Will transfer tonight... :)

**Comment by eflukx** [comment]
Ok gents, 

**Update:** Got payment/confirmation from  @raynet (1 set, paid), @patricklaf (1 set, paid), @northern (2 sets, paid) and @romsom (2 sets, paid). So I've got the people joining in for 6 sets (7 including myself)! ~~I will make the order at Mouser shortly!~~ ➡️ **ordered** Now the long wait ⏳ for the Ebay stuff starts... will keep you informed. 

**Note that there are three sets left, so if there's more interest, please let me know!**

Shipments from Mouser typically take a couple of days to come in (apart from the potentiometer that is back-ordered). Please be aware that some of the parts are NOS (new old stock), which I ordered from Ebay. Shipping will take some more time, I expect that will take ~1 - 1.5 months from now for the NOS parts to come in, after which I;m able to complete the sets. So a bit of patience is required.. ;) 

Ps. Please share your **shipping addresses** via the same email account as the PayPal account.

Ps. I can not guarantee the quality of the NOS parts. Typically these parts work just fine, the not-so-expensive parts like these tend to be OK and not counterfeits, but I have seen differently! Probably I'll just build one of the SB's myself (share some pictures in this thread) to see if all is working as expected. (hmm.. need to dust off an old ISA machine.. )

**Comment by raynet** [comment]
@eflukx I will take part too

just sent payment

**Comment by eflukx** [comment]
@northern, I got your payment! (seemed to have missed the email earlier). The amount seems correct, but PayPal has deducted a 7,64 fee. (Probably you selected "pay for goods" instead of "pay friends and family) I'll update the for the final amount when I ship the set out to you!

@romsom, @patricklaf  Please don't forget to send me your shipping details per email!

:octocat:  **All parts have been ordered** :octocat:

**Comment by northern** [comment]
@eflukx :man_facepalming: Ah, sorry mate. No worries, lets fix it up before you ship out.

**Comment by eflukx** [comment]
Hey, look what the cat dragged in...

![mouser_ics](https://user-images.githubusercontent.com/996191/61407511-32343880-a8de-11e9-9a02-124e8dc665a5.jpg)
![mouser_zakkies](https://user-images.githubusercontent.com/996191/61407512-32343880-a8de-11e9-945b-22b18b214de4.jpg)


It's a Mouse(r)!

**Comment by eflukx** [comment]
Gents, asked Roel to print the knobs for me. Thanks Roel!
I recall the original knob is black, but looks cool in White as well.. Anyone preference for white or black volume knob? Btw. as this is a FDM print, the "ridges" on the edge are not too well defined.. I'm afraid you'll just have to live with that.. 😉 

![knoppie_print](https://user-images.githubusercontent.com/996191/61480854-2fe1e500-a997-11e9-89a6-c2db1b5d452a.jpg)

**Comment by Sjaakafhaak** [comment]
Wel printer guy here.
I did some magic to the printer and as a matter of fact the look much nicer now.
And the ridges on the edge are pretty nice.

![IMG_2971](https://user-images.githubusercontent.com/6681276/61593361-d0761600-abde-11e9-8187-e07a159e8d99.JPG)

Also they are black like they should be :).

**Comment by eflukx** [comment]
Hi guys here's a short update of the parts delivered and stuff we're still waiting on...
~Most of the stuff is in already, but some items are back-ordered and some EBay (N)OS parts are still somewhere between China and Europe...~

**UPDATE: All electronics are in!**

Will try to find time this week to build one board and do the testing. Also need to make one-board-bundles for shipment obviously. If you just want your set shipped ASAP (and not want to await my building progress), give me a shout.

## Stuff i did receive
- [x] 3D Printed volume knobs
- [x] Most of the Mouser parts
- [x] Nice PCB's

## Stuff I plan to do myself before shipping
- [x] Make shippable bundles/sets of the parts
- [ ] Program the AT89S51 µC's
- [x] Build one full board 
- [x] Test above board

## Mouser back ordered items
- [x] S2 - Alps RK08H (10k dual potentiometer)
- [x] S1 - CUI SJ1-3553NG (3.5mm audio jack)

## EBay items
- [x] SAA1099 (CMS synthesizer)
- [x] YM3014B (Audio DAC)
- [x] TM3812 (OPL / Adlib FM-synth)
- [x] NE558 (Quad timer)
- [x] TEA2025B (Audio amp)

**Comment by LukeLovesRetro** [comment]
Hi, are there any kits left. I would also like one if possible to Australia.

**Comment by eflukx** [comment]
@LukeLovesRetro  
Hi there, yep I've got some left. Price for kit shipped to Australia will be Eur 95,-- (If forgot to include the tax/VAT in earlier calculation) 

Please send the money up front via Paypal address mentioned earlier in this thread, also don't forget to send me your shipping details via the same email address.

Cheers!

**Comment by Toesoe** [comment]
Hah, and I was just about to order all of the stuff myself when I found this thread.. @eflukx, do you have a set left, and if so, how much for the stuff + shipping within NL?

**Comment by eflukx** [comment]
@Toesoe Yep I have some kits left.. You came at the right time as essentially all components are in (except for a small counting mistake Mouser made, but will be fixed soon!) 

Kit price is EUR 60,-- w.o. shipping. So, say 67,-- incl. shipping within NL.

Ps. Ophalen in Den Haag / Rijswijk kan ook!  Krijg je ook koffie :)

**Comment by Toesoe** [comment]
@eflukx and others, after designating myself as a guinea pig I'm glad to say that (even with my crappy soldering) all parts work as intended, both the YM3812 and the SAA1099's! Had to program the AT89S51 with an old Arduino and ran into a few issues with my 166MHz AMD K6 being too fast for the card, but disabling L1 cache helped a lot (tip). It sounds excellent! https://imgur.com/a/48dfVgK

**Comment by eflukx** [comment]
@Toesoe Good to see you have your board working! Cool! 🥇 Thanks for your feed back on this. Let's consider this _the_ successful validation test: so I can cross that off my checklist ✔️  

@all Sorry for the loooong delay from my side. I have been quite busy coming back to work after summer vacation. Wasn't able to allot much time to the SnarkBarkers.. I _do_ have all your components sorted en ready to ship!

Somehow I'm having trouble with programming the 8051.. I tried different programmers: multiple USBASP's and an Arduino), at least 3 different versions of the USBASP firmware, multiple versions of AVRDude (real version hassle, apparently different versions have different config file layouts) and other programming tools, tried several different AT89S51's, different oscillator crystals, reset pin invertors yea/nea, different PC's, different bread boards... all to no avail. *sigh* The god's don't seem to be with me on this one.  :godmode: (Most frustrating is that all "other" peoples in the interwebz don't seem to have any problem with programming this baby...) (@toesoe will be in touch with you in your setup)

To prevent further delays, I'll start shipping soon (expect this weekend/beginning next week). Albeit with a virgin-like so some programming to do yourself.. ;) All that have paid up front I'll send you the tracking info by email.

**Comment by northern** [comment]
Hi @eflukx. Thanks for the update and apologies for the slow response. Sorry to hear you're having issues programming the 8051. If you feel it can't be done or you can't find the time then I guess you might as well send the components out. I don't really have any experience doing this kind of thing but I will give it a go. I think I still owe you some money for postage. Can you let me know how much that this so I can wire it to you?

Again, thanks for all you efforts so far. Very much appreciated!

**Comment by Toesoe** [comment]
For anyone looking to go the Arduino route, I sketched out the setup I used for as far as I remember it (it's been a couple of weeks :smile:). All the passives are available in @eflukx's kit.
![CQg6mCH](https://user-images.githubusercontent.com/38987846/65598150-119df400-df9b-11e9-858f-16edcf7760b1.png)
I used a cheap Chinese UNO clone that I've had lying around for, what, 6 years or so?
Add [the following](https://pastebin.com/Dkh2Yzix) to your avrdude.conf (it works for my version `6.3-20171130`), upload the ArduinoISP sketch and run avrdude with `-c stk500v1 -P <port> -p 89s51 -b 19200`. If it's detected properly, add `-U flash:w:sb.hex` to write the blob to the MC.

**Comment by raynet** [comment]
Got my kit, now just need to find time to build it :)

**Comment by patricklaf** [comment]
I have received the kit! Very nice printed knob and a lot of things to solder... Thank you all for making this possible.

**Comment by eflukx** [comment]
Actually haven't found the time to solder my own.. 😓 😉 💨 
@romsom You should have yours too.. 
@northern Your's was sent out at the same time, but probably takes a little longer to reach other end of the world.. 😉

**Comment by patricklaf** [comment]
I programmed my AT89S51 with my TL866CS without any problem.
If you have difficulty, you can send me your MCU, I will program it and send it back to you.

**Comment by romsom** [comment]
@eflukx Yes it's here! Thanks! :)
Haven't had the time to dig through it, though.
At least my trusty old Win95 machine is ready to be set up and program the AT89S51s ;)

**Comment by northern** [comment]
@eflukx Picked up the kits from the post office today. Thanks for taking great care with the shipping and  also thanks again for putting all this together. Hopefully I have some time over the xmas period to tinker with this a bit more. :)

**Comment by eflukx** [comment]
Cool, so everybody seem to have gotten their kits! That took a while to reach you @northern ! Happy it reached its destination 😄 
Apparently I'm the only one having troubles programming the 8051 in the end.. 😂 Will give it another try when I get 'round to build my own Snark Barker!
@all keep us posted (here) with the building result!
